saleor/graphql/product/mutations/products.py

```diff
@@ -739,6 +741,12 @@ def perform_mutation(cls, _root, info, **data):
         order_models.OrderLine.objects.filter(pk__in=line_pks).delete()
         if orders_id:
             recalculate_orders_task.delay(list(orders_id))
+
+        # update checkout quantity related to deleted lines
+        checkouts = checkout_models.Checkout.objects.filter(variant_id__in=variants_id)
```

Comment:
Can we figure out a different solution? This way has a big possibility to explode. We don't clean old checkout which it is easy to assume that we could have over 500k checkout objects. Assuming that 10% of this has the given variant assigned as a line will cause a huge amount of DB queries, `update_checkout_quantity` makes aggregation and save, so this action will cause around 100k queries. (Correct me if I am wrong) (this comment is related to all place where we call `update_checkout_quantity`)

Reply:
Maybe we should rewrite `update_checkout_quantity` to use `F` query on the whole CheckoutQueryset?

Reply:
I will try to find more optimised solution.